nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/QueryRecord.java

```diff
@@ -556,6 +566,23 @@ public int getRecordsRead() {
         }
     }
 
+    private SchemaPlus createRootSchema(final CalciteConnection calciteConnection) {
+        final SchemaPlus rootSchema = calciteConnection.getRootSchema();
+        // TODO: Document how to use RPATH functions in additionalDetails.html!
+        // TODO: Be sure to explain how/when to use each. For example, can just use RPATH when SELECTing the value but if using in a LIKE expression,
+        //       need a String so use RPATH_STRING
+        // TODO: Probably worth writing a blog post
+        rootSchema.add("RPATH", ScalarFunctionImpl.create(ObjectRecordPath.class, "eval"));
+        rootSchema.add("RPATH_STRING", ScalarFunctionImpl.create(StringRecordPath.class, "eval"));
```

Comment:
From the user perspective, is it better to have separate functions for the data types, or a single (perhaps overloaded) function that takes the data type as a string parameter?

Reply:
I could argue it either way, to be honest. However, it's not clear to me whether or not Calcite supports function overloading - or, if it does, how to accomplish it. It's also uncommon that the user will need to use anything other than the `RPATH` function but these are available just in case.